### PR TITLE
adoc: added list of changes from kubernetes 1.17. to 1.18

### PR DESCRIPTION
**Pull request by mssola**
I've added a new miscellanous page in which we document changes from 1.17 to 1.18 that might be relevant to our customers (e.g. API changes or some behavioral changes).

# Describe your changes

I've added into a new page the changes from Kubernetes 1.17 to 1.18. As for the behavioral changes, I've kept things simple, instead of writing larger descriptions as I initially intended (since it was becoming tedious to read and not that useful for customers in the end).

The changes described here are based on a presentation delivered by @mjura internally.

# Related Issues / Projects

Fixes SUSE/avant-garde#1420

I've decided not to base this PR from #756 since I'm not sure whether all the versions changed there are definitive.

### CODE REVIEW

File: adoc/admin-kubernetes-changes.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+== From Kubernetes 1.17 to 1.18
+
+This documentation page lists all the behavioral changes and API deprecations that will happen from Kubernetes 1.17 to 1.18. This is going to be relevant when migrating to {productname} 4.3.x, since it will bring Kubernetes 1.18.
+
+=== Architectural changes in {productname}
```
Comment (evrardjp) [inline adoc/admin-kubernetes-changes.adoc:5]: I am not sure this is very important to the deployers. I would rather have that as the release note in this repo, explaining in which part the hyperkube removal might impact them (mirroring of the image for airgapped).

I would remove this from here.

File: adoc/admin-kubernetes-changes.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+`hyperkube` has been removed from the Kubernetes project and is now community maintained. That being said, its future is uncertain, and it was no longer really needed for {productname}. Thus, we have decided to remove this binary starting from {productname} 4.3.0 and we are now providing binaries and container images for each Kubernetes component.
+
+This should not affect you directly: it will be transparent. That being said, if you relied on `hyperkube` specifically for some reason, take this into account.
+
+=== API deprecations
+
+The following APIs have been deprecated:
+
+- All resources under *apps/v1beta1* and *apps/v1beta2*: from now on use *apps/v1* instead.
```
Comment (evrardjp) [inline adoc/admin-kubernetes-changes.adoc:15]: nit: that sound very harsh, "from now on use" -> "please use" or "you must now use" ?

File: adoc/admin-kubernetes-changes.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+- link:https://github.com/kubernetes/enhancements/issues/853[#853] Configurable scale velocity for HPA.
+- link:https://github.com/kubernetes/enhancements/issues/1393[#1393] Provide OIDC discovery for service account token issuer.
+- link:https://github.com/kubernetes/enhancements/issues/1513[#1513] CertificateSigningRequest API.
+
+==== Scheduling
+
+- link:https://github.com/kubernetes/enhancements/issues/1451[#1451] Run multiple Scheduling Profiles [Alpha]
+- link:https://github.com/kubernetes/enhancements/issues/895[#895] Even pod spreading across failure domains [Beta]
+- link:https://github.com/kubernetes/enhancements/issues/1258[#1258] Add a configurable default Even Pod Spreading rule [Alpha]
```
Comment (evrardjp) [inline adoc/admin-kubernetes-changes.adoc:34]: should we mention alpha things?

Comment (mssola) [reply adoc/admin-kubernetes-changes.adoc:28]: I'd say so, as to give customers a heads up for incoming features for future versions.

**Comment by evrardjp** [comment]
Do not merge until 4.3.0

**Pull request merged**